lib/tasks/todo.rake

```diff
@@ -0,0 +1,7 @@
+namespace :todo do
+  desc "delete items older than 7 days"
+  task delete_item: :environment do
+    Item.where("created_at <= ? ", Time.now - 7.days).destroy_all
```

Comment:
Good to see the the rake task up and running.

What follows is an exercise in encapsulation and responsibility. The way the code is now is absolutely fine, but its a good opportunity to practice with these concepts.

Consider what the rake task knows about the inner workings of `Item`:

- That items expire after a week
- That the items db table has a `created_at` column

Now consider if the rake task needs to know these specific implementation details about `Item`.

Its usually beneficial to maintenance and readability of a codebase to keep the knowledge of implementation details localized to only the classes that need to know those details.

How could these implementation details be localized to just the `Item` class, so the rake task doesn’t know more than it needs to?

One refactoring technique that would help here is to move the code to a class method on `Item`:

```ruby
class Item …

  LIFETIME_IN_SECONDS = 7.days

  …
  def self.delete_expired
    where("created_at <= ? ", Time.now - LIFETIME_IN_SECONDS).destroy_all
  end
  …

end
```

The rake task now can limit its knowledge of how `Item` works to just knowing the name of this class method and calling it:

```ruby
task delete_item: :environment do
  Item.delete_expired
end
```

To help with maintenance and understanding of the codebase, a developer can now look at the `Item` class file and see all of its responsibilities in one file.